src/backend/distributed/metadata/dependency.c

```diff
@@ -1322,6 +1348,56 @@ ExpandCitusSupportedTypes(ObjectAddressCollector *collector, ObjectAddress targe
 }
 
 
+/*
+ * GetRelationRuleReferenceDependencyList returns the dependencies of the relation's
+ * internal rule dependencies.
+ */
+static List *
+GetRelationRuleReferenceDependencyList(Oid relationId)
+{
+	List *dependencyTupleList = GetPgDependTuplesForDependingObjects(RelationRelationId,
+																	 relationId);
+	List *nonInternalRuleDependencies = NIL;
```

Comment:
the name `nonInternalRuleDependencies` is slightly confusing. We first fetch internal dependencies of views (which are rules), then want to skip internal dependencies of the rules (as rules would create those dependencies automatically). 

Can you come up with a more descriptive name?